lib/level.js

```diff
@@ -0,0 +1,96 @@
+// CAUTION: This adapter does NOT handle encoding. an encoding mechanism like the encoding-down package will need to be included
+// Based on localStorage adapter in 05349a5
+
+var Gun   = ('undefined' !== typeof window) ? window.Gun : require('../gun');
+var debug = false;
+
+Gun.on('opt', function(ctx) {
+  var opt = ctx.opt;
+  var ev  = this.to;
+
+  if (debug) debug.emit('create');
+  if (ctx.once) return ev.next(ctx);
+
+  // Check if the given 'level' argument implements all the components we need
+  // Intentionally doesn't check for levelup explicitly, to allow different handlers implementing the same api
+  if (
+    (!opt.level) ||
+    ('object' !== typeof opt.level) ||
+    ('function' !== typeof opt.level.get) ||
+    ('function' !== typeof opt.level.put)
+  ) {
+    return;
+  }
+
+  ctx.on('put', function(msg) {
+   this.to.next(msg);
+
+    // Extract data from message
+    var put   = msg.put;
+    var soul  = put['#'];
+    var key   = put['.'];
+    var val   = put[':'];
+    var state = put['>'];
+
+    if (debug) debug.emit('put', soul, val);
+
+    // Fetch previous version
+    opt.level.get(soul, function(err, data) {
+      if (err && (err.name === 'NotFoundError')) err = undefined;
+      if (debug && err) debug.emit('error', err);
+      if (err) return;
+
+      // Unclear required transformation
+      data = Gun.state.ify(data, key, state, val, soul);
```

Comment:
you need to do merge/conflict resolution here in case incoming data is older than current data
mark says lib/mix.js is an example of merging

Reply:
I though the gun core handled the conflict resolution, not the adapters?

@amark How would I go about handling the conflict resolution here? `lib/mix.js` doesn't really contain self-explaining code.

Reply:
This is true tho I don't think any storage adapter properly does this (I still need to add it to RAD I think), so I'm fine with pulling without. The edge case is if disk has a write at time 7, we get a *no read* network write for something at time 6, and machine state is 9, so in-memory 6 > nothing, so it goes to storage which doesn't check 7 > 6.